src/dune_rules/install_rules.ml

```diff
@@ -179,15 +179,22 @@ end = struct
         in
         let modules_impl =
           List.concat_map installable_modules.impl ~f:(fun m ->
-              common m
-              @ List.filter_map Ml_kind.all ~f:(fun ml_kind ->
-                    let open Option.O in
-                    let+ cmt =
-                      Obj_dir.Module.cmt_file obj_dir m ~ml_kind
-                        ~cm_kind:(Ocaml Cmi)
-                    in
-                    (Cm_kind.Cmi, cmt))
-              |> set_dir m)
+              let cmt_files =
+                List.concat_map Ml_kind.all ~f:(fun ml_kind ->
+                    let open Lib_mode.Cm_kind in
+                    [ if_ (native || byte)
+                        ( Ocaml Cmi
+                        , Obj_dir.Module.cmt_file obj_dir m ~ml_kind
+                            ~cm_kind:(Ocaml Cmi) )
+                    ; if_ melange
+                        ( Melange Cmi
+                        , Obj_dir.Module.cmt_file obj_dir m ~ml_kind
+                            ~cm_kind:(Melange Cmi) )
+                    ]
+                    |> List.concat)
```

Comment:
Perhaps remove the duplication with:

```
List.concat_map [ (native || byte), Ocaml Cmi ; melange, Melange Cmi ] ~f:(...
```

Reply:
Done in https://github.com/anmonteiro/dune/pull/3/commits/27905d18419beb0a454450c5f04783be0d74beb4.